api/campaigns/index.js

```diff
@@ -90,26 +92,31 @@ router.get('/camp/:id', (req, res) => {
 
 router.post('/', mw.upload.single('photo'), async (req, res) => {
   const { location } = req.file;
-
   const postCamp = {
-    ...req.body,
+    users_id:req.body.users_id,
+    camp_name:req.body.camp_name,
+    camp_desc:req.body.camp_desc,
+    camp_cta:req.body.camp_cta,
+    urgency:req.body.urgency,
+    is_archived:req.body.is_archived,
     camp_img: location
   };
-
   try {
     const newCamps = await Camp.insert(postCamp);
-    if (newCamps) {
-      log.info(newCamps);
+    const newSkilledImpactRequests =
+        await SkilledImpactRequests.insertSkilledImpactRequests(req.body.skilled_impact_requests, newCamps.camp_id);
+    if (newCamps&&newSkilledImpactRequests) {
+      log.info('inserted campaign including skilled impact requests', newCamps, newSkilledImpactRequests);
       res.status(201).json({ newCamps, msg: 'Campaign added to database' });
     } else if (
       !postCamp.camp_img ||
       !postCamp.camp_name ||
       !postCamp.camp_desc ||
-      !postCamp.camp_cta
+      !postCamp.camp_cta || !req.body.skilled_impact_requests
     ) {
       log.info('no data');
       res.status(404).json({
-        msg: 'You need campaign image, campaign name, and campaign description'
+        msg: 'You need campaign image, campaign name, campaign description, and skilled impact requests'
```

Comment:
Changing client-facing error messages is usually not a great idea. We should refactor our error handling some time soon so we can make changes like this.

@sumeet-bansal Any idea if mobile relies on this error message?

Reply:
Why's changing client-facing error messages an issue? But we definitely need to standardize responses (e.g. including `error: null` in successful responses) across the app.

Reply:
There needs to be some immutable field of every error message so the frontend can pattern match on every possible error from a route, to show the user error messages or whatnot. 

We should define immutable error codes (like `code: "MISSING_FIELDS"`), and we can put these human-readable, mutable fields alongside it.